src/composites/ConfigurationContainer/ImportanceAdjustment/AdjsutmentTable.tsx

```diff
@@ -0,0 +1,210 @@
+import { Button, Flex, Text, HoverCard, Link, Strong, Table, Box, Card, Avatar } from "@radix-ui/themes";
+import { ResetIcon, DownloadIcon } from "@radix-ui/react-icons";
+import SingleTableRow from "./SingleRowInTable";
+import { Profile } from '../../../types';
+
+
+interface Weights {
+    [key: string]: number;
+}
+
+interface TQIEntry {
+    weights: Weights;
+}
+
+interface AdjustmentTableProps {
+    selectedProfile?: Profile[];
+    isProfileApplied: boolean;
+    onResetApplied: () => void; // New prop for reset handler
+}
+
+
+export const AdjustmentTable: React.FC<AdjustmentTableProps> = ({ selectedProfile, isProfileApplied, onResetApplied }) => {
+
+    const dataset = useAtomValue(State.dataset);
+    if (!dataset) return null;
+
+     // Define a function to calculate initial weights
+     const getInitialWeights = (useDataset: boolean = false) => {
+        let weights: Weights = {};
+        if (selectedProfile && selectedProfile.length > 0 && !useDataset) {
+            // Use weights from the first selected profile
+            const profileWeights = selectedProfile[0].importance;
+            Object.entries(profileWeights).forEach(([aspect, importance]) => {
+                weights[aspect] = importance;
+            });
+        } else {
+            // Fallback to dataset weights
+            Object.entries(dataset.factors.tqi).forEach(([_, tqiEntry]) => {
+                const entry = tqiEntry as TQIEntry;
+                Object.assign(weights, entry.weights);
+            });
+        }
+        return weights;
+    };
+
+    // State for slider values
+    const [sliderValues, setSliderValues] = useState<Weights>(getInitialWeights);
+
+    // Reset sliders to initial weights (dataset or selected profile)
+    const resetAllAdjustments = () => {
+        setSliderValues(getInitialWeights(true)); // Force use of dataset weights
+        onResetApplied(); // Signal to parent that reset was triggered
+    };
+
+   // Calculate total importance for normalization
+   const totalImportance = useMemo(() => Object.values(sliderValues).reduce((sum, importance) => sum + importance, 0), [sliderValues]);
+
+
+    // Recalculate weights based on slider adjustments
+    const recalculatedWeights = useMemo(() => {
+        const newWeights: Weights = {};
+        Object.keys(sliderValues).forEach(name => {
+            newWeights[name] = sliderValues[name] / totalImportance;
+        });
+        return newWeights;
+    }, [sliderValues, totalImportance]);
+
+
+    // Handle slider value change
+    const handleSliderChange = (name: string, newImportance: number) => {
+        setSliderValues(prev => ({ ...prev, [name]: newImportance }));
+    };
+
+    // Update sliderValues when selectedProfile changes
+    if (isProfileApplied) {
+        const newInitialWeights = getInitialWeights();
+        if (JSON.stringify(newInitialWeights) !== JSON.stringify(sliderValues)) {
+            setSliderValues(newInitialWeights);
```

Comment:
hmm this will cause problems, since `isProfileApplied` comes from props and this state change relies on changes to props. This can run into an infinite loop. One solution is to pass slider values from the parent directly. Is there any reason why these cannot be derived directly in a `useMemo`?

Reply:
updated with useMemo